numpy/_core/src/multiarray/descriptor.c

```diff
@@ -726,32 +726,34 @@ _convert_from_list(PyObject *obj, int align)
 static PyArray_Descr *
 _convert_from_commastring(PyObject *obj, int align)
 {
-    PyObject *listobj;
+    PyObject *parsed;
     PyArray_Descr *res;
-    PyObject *_numpy_internal;
+    static PyObject *_numpy_internal = NULL;
     assert(PyUnicode_Check(obj));
-    _numpy_internal = PyImport_ImportModule("numpy._core._internal");
     if (_numpy_internal == NULL) {
-        return NULL;
+        _numpy_internal = PyImport_ImportModule("numpy._core._internal");
```

Comment:
Would be a bit nicer via:
```
npy_cache_import("numpy._core._internal", "_commastring", &_commastring);
```
And calling it with:
```
PyObject_CallOneArg(_commanstring, obj);
```

Reply:
Yes, of course, should have thought of that!